src/map/packet_system.cpp

```diff
@@ -6178,32 +6176,53 @@ void SmallPacket0x0DB(map_session_data_t* const PSession, CCharEntity* const PCh
 {
     TracyZoneScoped;
 
-    auto oldMenuConfigFlags = PChar->menuConfigFlags.flags;
-    auto oldChatFilterFlags = PChar->chatFilterFlags;
-    auto oldLanguages       = PChar->search.language;
+    // https://github.com/atom0s/XiPackets/tree/main/world/client/0x00DB
+    struct packet_c2s_0DB_t
+    {
+        uint16_t id : 9;
+        uint16_t size : 7;
+        uint16_t sync;
+        uint8_t  unknown04;    // Set to 0.
+        uint8_t  unknown05;    // Set to 0.
+        uint8_t  Kind;         // The packet kind.
+        uint8_t  padding00;    // Padding; unused.
+        uint32_t ConfigSys[3]; // The players current PTR_pGlobalNowZone->ConfigSys values.
+        uint32_t padding01[4]; // Padding; unused. (Space for future information?)
+        uint32_t Param;        // Packet parameter.
+    } packet = {};
 
-    // Extract the system filter bits and update MenuConfig
-    const uint8 systemFilterMask = (NFLAG_SYSTEM_FILTER_H | NFLAG_SYSTEM_FILTER_L) >> 8;
-    PChar->menuConfigFlags.byte2 &= ~systemFilterMask;
-    PChar->menuConfigFlags.byte2 |= data.ref<uint8>(0x09) & systemFilterMask;
+    std::memcpy(&packet, data, sizeof(packet_c2s_0DB_t));
 
-    PChar->chatFilterFlags = data.ref<uint64>(0x0C);
+    uint32_t oldPlayerConfig = {};
+    uint32_t oldChatFilter1  = {};
+    uint32_t oldChatFilter2  = {};
 
-    PChar->search.language = data.ref<uint8>(0x24);
+    std::memcpy(&oldPlayerConfig, &PChar->playerConfig, sizeof(uint32_t));
+    std::memcpy(&oldChatFilter1, &PChar->playerConfig.MassageFilter, sizeof(uint32_t));
+    std::memcpy(&oldChatFilter2, &PChar->playerConfig.MassageFilter2, sizeof(uint32_t));
 
-    if (oldMenuConfigFlags != PChar->menuConfigFlags.flags)
+    if (packet.Kind == 1)
```

Comment:
Can these magic numbers get enum'd, or some comments?

Reply:
Enums for things like `kind` and `param` aren't really knowable because they got compiled into numbers, but comments aren't a problem